gems/pending/appliance_console/date_time_configuration.rb

```diff
@@ -0,0 +1,118 @@
+require "appliance_console/logging"
+require 'appliance_console/prompts'
+
+module ApplianceConsole
+  class DateTimeConfiguration
+    DATE_REGEXP   = /^(2[0-9]{3})-(0?[1-9]|1[0-2])-(0?[1-9]|[12][0-9]|3[01])/
+    DATE_PROMPT   = "current date (YYYY-MM-DD)".freeze
+    TIME_REGEXP   = /^(0?[0-9]|1[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])/
+    TIME_PROMPT   = "current time in 24 hour format (HH:MM:SS)".freeze
+
+    attr_accessor :new_date, :new_time, :manual_time_sync
+
+    include ApplianceConsole::Logging
+
+    def initialize
+      @new_date         = nil
+      @new_time         = nil
+      @manual_time_sync = false
+    end
+
+    def activate
+      say("Applying time configuration...")
+      establish_auto_sync &&
+        configure_date_time
+    end
+
+    def ask_questions
+      ask_establish_auto_sync
+      ask_for_date &&
+        ask_for_time &&
+        confirm
+    end
+
+    def ask_establish_auto_sync
+      say("Automatic time synchronization must be disabled to manually set date or time\n\n")
+
+      @manual_time_sync = agree(<<-EOL)
+Yes to disable Automatic time synchronization and prompt for date and time.
+ No to enable  Automatic time synchronization.  (Y/N):
```

Comment:
Not sure on the wording, or the feel of where the Y/N prompt falls, but we can debate that later... cc @chessbyte